### PR TITLE
Add tdengine to telemetry service

### PR DESCRIPTION
**Pull request by geraldkohn**
**What this PR does / why we need it**:
add tdengine to telemetry service
**Will this PR make the community happier**? 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [x] unit test
- [x] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
The PR includes writing and querying of tdengine data (without binding an http handler). 
The service distinguishes devices by HTTP request headers, 
with the key of the device name being "device_name" and the key of the event categories being "device_tag".

The following are tentatively determined structured data formats:
type StructureData struct {
	Timestamp  time.Time `json:"ts"`         // event time stamp
	Data       string    `json:"data"`       // event content
	Tag        string    `json:"tg"`         // event categories
}
```

### CODE REVIEW

**Review by tomqin93 — CHANGES_REQUESTED** [review]
@geraldkohn please check in the data design for this PR

File: pkg/telemetryservice/sql/tdengine/tdengine.go [hunk 1]
```diff
@@ -12,23 +12,54 @@ import (
...
 	if err != nil {
-		logger.Errorf("Error to Connect to tdengine, error %v", err.Error())
+		logger.Errorf("Error to Connect to tdengine, error %s", err.Error())
 		return err
 	}
 
-	err = db.insertDataToDB(ctx, rawData)
+	deviceInfo := &DeviceInfo{}
+	if name, exists := header[DeviceNameHeaderField]; exists {
+		deviceInfo.Name = name[0]
```
Comment (kris21he) [inline pkg/telemetryservice/sql/tdengine/tdengine.go:48]: should to check if len(name) > 0?

File: pkg/telemetryservice/sql/tdengine/tdengine.go [hunk 2]
```diff
@@ -12,23 +12,54 @@ import (
...
+	} else {
+		logger.Infof("Error to get device name from http header")
+		deviceInfo.Name = "default_device_name"
+	}
+	if tag, exists := header[EventTagHeaderField]; exists {
+		deviceInfo.Tag = tag[0]
```
Comment (kris21he) [inline pkg/telemetryservice/sql/tdengine/tdengine.go:54]: check len(tag) > 0

File: pkg/telemetryservice/sql/tdengine/tdengine.go [hunk 3]
```diff
@@ -12,23 +12,54 @@ import (
...
+		logger.Errorf("Error to Connect to tdengine, error %s", err.Error())
 		return err
 	}
 
-	err = db.insertDataToDB(ctx, rawData)
+	deviceInfo := &DeviceInfo{}
+	if name, exists := header[DeviceNameHeaderField]; exists {
+		deviceInfo.Name = name[0]
+	} else {
+		logger.Infof("Error to get device name from http header")
```
Comment (kris21he) [inline pkg/telemetryservice/sql/tdengine/tdengine.go:50]: if this is an error should use logger.Errorf,
else just logger.Infof("empty device name from http header")

File: pkg/telemetryservice/sql/tdengine/tdengine.go [hunk 4]
```diff
@@ -12,23 +12,54 @@ import (
...
+		deviceInfo.Name = "default_device_name"
+	}
+	if tag, exists := header[EventTagHeaderField]; exists {
+		deviceInfo.Tag = tag[0]
+	} else {
+		logger.Infof("Error to get device tag from http header")
```
Comment (kris21he) [inline pkg/telemetryservice/sql/tdengine/tdengine.go:56]: ditto

File: pkg/telemetryservice/sql/tdengine/tdengine.go [hunk 5]
```diff
@@ -12,23 +12,54 @@ import (
...
 		return err
 	}
 
-	err = db.insertDataToDB(ctx, rawData)
+	deviceInfo := &DeviceInfo{}
+	if name, exists := header[DeviceNameHeaderField]; exists {
+		deviceInfo.Name = name[0]
+	} else {
+		logger.Infof("Error to get device name from http header")
+		deviceInfo.Name = "default_device_name"
```
Comment (kris21he) [inline pkg/telemetryservice/sql/tdengine/tdengine.go:51]: use string const

File: pkg/telemetryservice/sql/tdengine/tdengine.go [hunk 6]
```diff
@@ -67,3 +118,40 @@ func (db *DBHelper) insertDataToDB(ctx context.Context, rawData []byte) error {
 func constructTDengineUri(sqlcs *v1alpha1.SQLConnectionSetting) string {
 	return fmt.Sprintf("%s:%s@http(%s)/%s", *sqlcs.UserName, *sqlcs.Secret, *sqlcs.ServerAddress, *sqlcs.DBName)
 }
+
+func (db *DBHelper) query(querySql string) ([]*StructureData, error) {
+	StructureDatas := make([]*StructureData, 0)
+	rows, err := db.DB.Query(querySql)
+	if err != nil {
+		logger.Errorf("Error to query data from tdengine, sql code: %s", querySql)
```
Comment (kris21he) [inline pkg/telemetryservice/sql/tdengine/tdengine.go:126]: logger.Errorf("Error to query data from tdengine, sql: %s, error: %s", querySql, err.Error())

File: pkg/telemetryservice/sql/tdengine/tdengine.go [hunk 7]
```diff
@@ -67,3 +118,40 @@ func (db *DBHelper) insertDataToDB(ctx context.Context, rawData []byte) error {
 func constructTDengineUri(sqlcs *v1alpha1.SQLConnectionSetting) string {
 	return fmt.Sprintf("%s:%s@http(%s)/%s", *sqlcs.UserName, *sqlcs.Secret, *sqlcs.ServerAddress, *sqlcs.DBName)
 }
+
+func (db *DBHelper) query(querySql string) ([]*StructureData, error) {
+	StructureDatas := make([]*StructureData, 0)
```
Comment (kris21he) [inline pkg/telemetryservice/sql/tdengine/tdengine.go:123]: dataRecords := make([]*StructureData, 0)

File: pkg/telemetryservice/sql/tdengine/tdengine.go [hunk 8]
```diff
@@ -67,3 +118,40 @@ func (db *DBHelper) insertDataToDB(ctx context.Context, rawData []byte) error {
...
+		if err != nil {
+			logger.Errorf("Error to scan result into structureData")
+			return nil, err
+		}
+		StructureDatas = append(StructureDatas, s)
+	}
+	return StructureDatas, nil
+}
+
+func (db *DBHelper) queryFromDeviceName(devicename string) ([]*StructureData, error) {
```
Comment (kris21he) [inline pkg/telemetryservice/sql/tdengine/tdengine.go:141]: actually I don't think we have the requirement of query data from telemetry service, do we?

**Pull request closed without merging**